utils/Retail/Confirm/onConfirm.ts

```diff
@@ -226,27 +227,43 @@ export const checkOnConfirm = (data: any) => {
       )
     }
 
+    try {
+      logger.info(`Storing delivery fulfillment if provided in ${constants.ON_CONFIRM}`)
+      const deliveryFulfillment = on_confirm.fulfillments.filter((fulfillment: any) => fulfillment.type === 'Delivery')
+      const { start, end } = deliveryFulfillment[0]
+      const startRange = start.time.range
+      const endRange = end.time.range
+
+      if (!startRange || !endRange) {
+        onCnfrmObj[
+          `fulfillment.${[deliveryFulfillment.id]}.range`
```

Comment:
providing time ranges is optional in on_confirm. Could you provided later once the order is packed or picked

Reply:
Done